internal/provider/resource_agent.go

```diff
@@ -57,6 +59,9 @@ func (*Agent) Schema(ctx context.Context, req resource.SchemaRequest, resp *reso
 			"id": schema.StringAttribute{
 				MarkdownDescription: "The ID of the Agent.",
 				Required:            true,
+				PlanModifiers: []planmodifier.String{
```

Comment:
Does this mean that the agent with the old id is removed?

Reply:
Yes, `RequiresReplace` indicates that this attribute can't be updated in-place and instead the old one needs to be deleted and a new resource needs to be created.

We also use this for example for applications: https://github.com/humanitec/terraform-provider-humanitec/blob/main/internal/provider/resource_application.go#L69